### PR TITLE
support partition format

### PR DESCRIPTION
**Pull request by skydoorkai**
To support partition format in `data_reader_params` as `partition=dt=20191130`.

### CODE REVIEW

File: elasticdl/python/common/model_utils.py [hunk 1]
```diff
@@ -38,8 +38,14 @@ def get_dict_from_params_str(params_str):
         kvs = params_str.split(";")
         params_dict = {}
         for kv in kvs:
-            k, v = kv.strip().split("=")
-            params_dict[k] = eval(v)
+            splitted = kv.strip().split("=")
+            k = splitted[0]
+            # if there is '=' in value, need to restore it.
```
Comment (terrytangyuan) [inline elasticdl/python/common/model_utils.py:43]: Add a test case?

File: elasticdl/python/master/master.py [hunk 2]
```diff
@@ -37,13 +38,21 @@ def _make_task_dispatcher(
...
     num_epochs,
+    data_reader_params,
 ):
     # TODO: Support any subclasses of `AbstractDataReader`
     # and support passing specified parameters to the constructor
     def _maybe_create_shards(data_origin):
+        wkargs = get_dict_from_params_str(data_reader_params)
+        if wkargs and "partition" in wkargs:
+            partition = wkargs["partition"]
+        else:
```
Comment (terrytangyuan) [inline elasticdl/python/master/master.py]: wkargs.get(key, default_value) is probably what you wanted?

File: elasticdl/python/common/model_utils.py (hunk 1 above)
Comment (skydoorkai) [reply elasticdl/python/common/model_utils.py:43]: Added.

File: elasticdl/python/master/master.py (hunk 2 above)
Comment (skydoorkai) [reply elasticdl/python/master/master.py]: Rewrote.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**